### PR TITLE
increase retransmit shreds received cache size

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Retransmit shreds received cache can overflow with supported slot sizes.

#### Summary of Changes
- Increase packet cache size based on max number of shreds per slot.
- Add metric to count number of times an entry is evicted from the LRU shred cache to detect overflow.

Fixes #

### CODE REVIEW

File: core/src/retransmit_stage.rs [hunk 1]
```diff
@@ -165,8 +182,7 @@ fn maybe_reset_shreds_received_cache(
     packet_hasher: &mut PacketHasher,
```
Comment (bw-solana) [inline core/src/retransmit_stage.rs:182]: Why do we need to reset the shreds received cache at all?

File: core/src/retransmit_stage.rs [hunk 2]
```diff
@@ -42,7 +47,10 @@ use {
 };
 
 const MAX_DUPLICATE_COUNT: usize = 2;
-const DEFAULT_LRU_SIZE: usize = 10_000;
+const SHREDS_RECEIVED_CACHE_UPDATE_INTERVAL: Duration = Duration::from_secs(1);
+const DEFAULT_LRU_SIZE: usize =
+    (MAX_DATA_SHREDS_PER_SLOT as u64 * 2 * SHREDS_RECEIVED_CACHE_UPDATE_INTERVAL.as_millis() as u64
```
[the remark this answers is not in the document]
Comment (jbiseda) [reply core/src/retransmit_stage.rs]: I rearranged this and added a comment. There isn't a const for `MAX_CODING_SHREDS_PER_SLOT`. I could introduce this if it makes sense @behzadnouri

File: core/src/retransmit_stage.rs [hunk 3]
```diff
@@ -42,7 +47,10 @@ use {
 };
 
 const MAX_DUPLICATE_COUNT: usize = 2;
-const DEFAULT_LRU_SIZE: usize = 10_000;
+const SHREDS_RECEIVED_CACHE_UPDATE_INTERVAL: Duration = Duration::from_secs(1);
+const DEFAULT_LRU_SIZE: usize =
```
[the remark this answers is not in the document]
Comment (jbiseda) [reply core/src/retransmit_stage.rs]: done

File: core/src/retransmit_stage.rs (hunk 1 above)
Comment (jbiseda) [reply core/src/retransmit_stage.rs:185]: Clearing the cache will clear the duplicate versions. It seems safer to keep the existing behavior.

File: core/src/retransmit_stage.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (behzadnouri) [reply core/src/retransmit_stage.rs]: I think we can just use a large literal value here; something like `1 << 20`.

**Review by bw-solana — APPROVED** [review]

File: core/src/retransmit_stage.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbiseda) [reply core/src/retransmit_stage.rs]: > I think we can just use a large literal value here; something like `1 << 20`.

@behzadnouri Do you think I should just increase the value and skip adding a metric?

[the remark this answers is not in the document]
Comment (behzadnouri) [reply core/src/retransmit_stage.rs]: yeah, that should be fine

**Review by behzadnouri — APPROVED** [review]

**Pull request merged**